packages/scandipwa/src/component/Form/Form.component.js

```diff
@@ -42,21 +42,32 @@ export class Form extends PureComponent {
         id: ''
     };
 
-    static updateChildrenRefs(props) {
+    static updateChildrenRefs(props, state = {}) {
         const { children: propsChildren } = props;
+        const { refMap: refMapState = {} } = state;
 
         const refMap = {};
 
         const children = Form.cloneChildren(
             propsChildren,
             (child) => {
                 const { props: { name } } = child;
+                const { message } = Object.keys(refMapState).length
+                    ? Form.validateField(child, refMapState)
+                    : {};
```

Comment:
Re-calculating Object.keys during each iteration of this callback is redundant. Please, calculate it only once.

Reply:
Forgot to mention, that this is just a quick issue fix improvement, as time was limited, overall the whole logic for the component does some conflicts by itself.

`...Form.cloneAndValidateChildren(children, refMap),
...Form.updateChildrenRefs(props, state)`

Like these, both methods return children's prop, so the second overrides the first one, and really hard to understand what's even happening there.